scripts/kfctl.sh

```diff
@@ -26,80 +26,103 @@ DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
 source "${DIR}/util.sh"
 source "${DIR}/gke/util.sh"
 source "${DIR}/util-minikube.sh"
+INPUT=''
+FORMAT=''
+export KUBEFLOW_COMPONENTS=${DEFAULT_KUBEFLOW_COMPONENTS:-'"ambassador","jupyter","centraldashboard","tf-job-operator","pytorch-operator","spartakus","argo","pipeline"'}
+export KUBEFLOW_EXTENDEDINFO=${KUBEFLOW_EXTENDEDINFO:-false}
+
+writeEnv() {
+  echo -e $INPUT | envsubst $FORMAT > ${ENV_FILE}
+}
 
 createEnv() {
   # Check if there is a file env.sh
   # If there is source it otherwise create it.
   # this ensures all relevant environment variables are persisted in
   # a file for consistency across runs.
-  echo PLATFORM=${PLATFORM} >> ${ENV_FILE}
+
   DEFAULT_KUBEFLOW_REPO="$(cd "${DIR}/.." > /dev/null && pwd)"
   # Remove trailing slash from the repo.
   KUBEFLOW_REPO=${KUBEFLOW_REPO%/}
-  echo KUBEFLOW_REPO=${KUBEFLOW_REPO:-"${DEFAULT_KUBEFLOW_REPO}"} >> ${ENV_FILE}
-  echo KUBEFLOW_VERSION=${KUBEFLOW_VERSION:-"master"} >> ${ENV_FILE}
-  echo KUBEFLOW_KS_DIR=${KUBEFLOW_KS_DIR:-"$(pwd)/ks_app"} >> ${ENV_FILE}
-  echo KUBEFLOW_DOCKER_REGISTRY=${KUBEFLOW_DOCKER_REGISTRY:-""} >> ${ENV_FILE}
-  echo DOCKER_REGISTRY_KATIB_NAMESPACE=${DOCKER_REGISTRY_KATIB_NAMESPACE:-""} >> ${ENV_FILE}
 
+  INPUT+='PLATFORM=$PLATFORM\nKUBEFLOW_REPO=$KUBEFLOW_REPO\nKUBEFLOW_VERSION=$KUBEFLOW_VERSION\n'
```

Comment:
What is this for?

Reply:
it does something like envsubst and makes it easy to update the env.sh file. Sort of a best practice in bash to create/update env files

Reply:
Makes sense.

nit 1: Add a comment explaining that
nit 2: Code will be a bit more readable if its 1 variable per line; we can leave that as a TODO for the future.